### PR TITLE
orchagent: Adding orchagent reinitializing logic

### PR DESCRIPTION
**Pull request by stcheng**

### CODE REVIEW

File: orchagent/main.cpp [hunk 1]
```diff
@@ -205,6 +209,15 @@ int main(int argc, char **argv)
     }
 
     try {
+        if (reinit)
+        {
+            SWSS_LOG_NOTICE("Start orchagent reinitialization");
+            orchDaemon->reinit();
+            /* After reinitialization, a second initialize_switch function is called
+             * to notify the ASIC that reinitialization is done. */
+            sai_switch_api->initialize_switch(0, "", "", &switch_notifications);
```
Comment (kcudnik) [inline orchagent/main.cpp]: take a look at sairedis implementation on initialize you need to pass pecific string to tell that

File: orchagent/main.cpp [hunk 2]
```diff
@@ -147,8 +151,22 @@ int main(int argc, char **argv)
...
+         * the old ASIC view to the new one. */
+        SWSS_LOG_NOTICE("sai_switch_api: RE-initializing switch");
+        status = sai_switch_api->initialize_switch(0, "0xb850",
+                "sai_switch_view", &switch_notifications);
+    }
+    else
+    {
+        SWSS_LOG_NOTICE("sai_switch_api: initializing switch");
+        status = sai_switch_api->initialize_switch(0, "0xb850",
+                "", &switch_notifications);
```
Comment (kcudnik) [inline orchagent/main.cpp]: you are missing here also "switch" or "compile"

Comment (stcheng) [inline orchagent/main.cpp]: fixed

**Comment by kcudnik** [comment]
Ack

**Comment by stcheng** [comment]
Will be merged after finishing all the other core features.

**Comment by kcudnik** [comment]
Ack

**Comment by stcheng** [comment]
This is closed temporarily and waiting for features to be ready and then get updated.

**Pull request closed without merging**